src/modules/notification-settings/notification-settings.service.ts

```diff
@@ -0,0 +1,78 @@
+import { NotificationSettingsDto } from './dto/notification-settings.dto';
+
+@Injectable()
+export class NotificationSettingsService {
+  private readonly logger = new Logger(NotificationSettingsService.name);
+
+  constructor(
+    @InjectRepository(NotificationSettings)
+    private notificationSettingsRepository: Repository<NotificationSettings>
+  ) {}
+
+  async create(notificationSettingsDto: any, user_id: string): Promise<any> {
+    try {
+      const existingSettings = await this.notificationSettingsRepository.findOne({
+        where: { user_id },
+      });
+
+      if (existingSettings) {
+        Object.assign(existingSettings, notificationSettingsDto);
+        return this.notificationSettingsRepository.save(existingSettings);
+      }
+
+      const newSettings = this.notificationSettingsRepository.create({ ...notificationSettingsDto, user_id });
+      return this.notificationSettingsRepository.save(newSettings);
+    } catch (error) {
+      this.logger.error('Error creating notification settings:', error);
+      throw new BadRequestException(error.message || 'Failed to create notification settings');
+    }
+  }
+
+  async findNotificationSettingsByUserId(user_id: string): Promise<NotificationSettings> {
+    try {
+      const defaultNotificationSettings = {
+        email_notification_activity_in_workspace: true,
+        email_notification_always_send_email_notifications: false,
+        email_notification_email_digest: true,
+        email_notification_announcement_and_update_emails: true,
+        slack_notifications_activity_on_your_workspace: false,
+        slack_notifications_always_send_email_notifications: true,
+        slack_notifications_announcement_and_update_emails: false,
+      };
+
+      let settings = await this.notificationSettingsRepository.findOne({ where: { user_id } });
+      if (!settings) {
+        settings = await this.create(defaultNotificationSettings, user_id);
+      }
+
+      return settings;
+    } catch (error) {
+      this.logger.error('Error finding notification settings by user ID:', error);
+      throw new BadRequestException(error?.message || 'Failed to fetch notification settings');
+    }
+  }
+  async updateNotificationSettings(
+    user_id: string,
+    notificationSettingsDto: Partial<NotificationSettingsDto>
+  ): Promise<Partial<NotificationSettingsDto>> {
+    try {
+      const settings = await this.notificationSettingsRepository.findOne({ where: { user_id } });
+      if (!settings) {
+        throw new NotFoundException('User settings not found');
+      }
+      Object.assign(settings, notificationSettingsDto);
+      await this.notificationSettingsRepository.save(settings);
+      return { ...notificationSettingsDto };
+    } catch (error) {
+      this.logger.error('Error updating notification settings:', error);
+      if (error instanceof NotFoundException) {
+        throw error;
+      }
+      throw new BadRequestException('Invalid data in the request body');
```

Comment:
Your catch block can also catch dB errors. So this would send a false response. Reevaluate the badRequest and InternalServerExceptions